theano/tensor/nnet/abstract_conv.py

```diff
@@ -1484,6 +1584,102 @@ def conv3d_grad_wrt_weights(input,
     return gradWeight_op(input, output_grad, filter_shape[-3:])
 
 
+def causal_conv(input,
```

Comment:
I think it's better to rename this to `causal_conv1d`, for consistency with the other methods.